addons/mail/static/src/models/notification_group/notification_group.js

```diff
@@ -62,11 +62,51 @@ function factory(dependencies) {
 
         /**
          * @override
+         * @private
          */
         static _createRecordLocalId(data) {
             return `${this.modelName}_${data.id}`;
         }
 
+        /**
+         * Compute the most recent date inside the notification messages.
+         *
+         * @private
+         */
+        _computeDate() {
+            const groupDate = moment.max(this.notifications.map(notification => {
+                if (notification.message && notification.message.date) {
+                    return notification.message.date;
+                }
+            }));
+            return groupDate;
+        }
+
+        /**
+         * Compute the threadId of the notification group if all the message
+         * have the same origin.
+         *
+         * @private
+         */
+        _computeResId() {
+            const notificationsThreadIds = this.notifications.map(notification => notification.message.originThread.id);
+            const threadIds = new Set(notificationsThreadIds);
+            /**
+             * keep res_id only if all notifications are for the same record
+             * set null if multiple records are present in the group
+             */
+            return (threadIds.size == 1) ? notificationsThreadIds[0] : null;
```

Comment:
Actually now that I think about it, I think `res_id` is only used here to compute `thread` (to double check).
So it might actually be much easier to just compute `thread` directly in a similar way, and get rid of `res_id`.